internal/datastore/proxy/caching.go

```diff
@@ -80,7 +80,7 @@ func (r *nsCachingReader) ReadNamespace(
 		loadedRaw, err, _ = r.p.readNsGroup.Do(nsRevisionKey, func() (any, error) {
 			// sever the context so that another branch doesn't cancel the
 			// single-flighted namespace read
-			loaded, updatedRev, err := r.Reader.ReadNamespace(datastore.SeparateContextWithTracing(ctx), nsName)
+			loaded, updatedRev, err := r.Reader.ReadNamespace(SeparateContextWithTracing(ctx), nsName)
```

Comment:
Why are we leaving this one here?

Reply:
We actually don't want to cancel this -- it explicitly has a test to assert this behavior.